src/Servers/HttpSys/src/DelegationRule.cs

```diff
@@ -34,8 +37,20 @@ internal DelegationRule(string queueName, string urlPrefix, ILogger logger)
         /// <inheritdoc />
         public void Dispose()
         {
-            Queue.UrlGroup?.Dispose();
-            Queue?.Dispose();
+            if (_disposed)
+            {
+                return;
+            }
+
+            _disposed = true;
+
+            try
+            {
+                _sourceQueueUrlGroup.UnSetDelegationProperty(Queue, throwOnError: false);
+            }
+            catch (ObjectDisposedException) { /* Server may have been shutdown */ }
+            Queue.UrlGroup.Dispose();
+            Queue.Dispose();
```

Comment:
This pattern for explicitly disposing a nested resource is really weird, I'm not sure why we did it that way. Queue should know if it owns the UrlGroup and if it should dispose it. Can you fix that here?

Reply:
I'm guessing this in artifact of the HttpListener code that's had this for a while

https://github.com/dotnet/aspnetcore/blob/master/src/Servers/HttpSys/src/HttpSysListener.cs#L258-L275

It looks weird since it looks like the Queue owns the UrlGroup and ergo is responsible for disposing it, but it in reality it's that the delegation rule owns the both the UrlGroup and the RequestQueue. The Queue just happens to have a reference to it's peer that we use to look up. It'll look a lot less weird if I stashed both the queue and the UrlGroup on the delegation rule.

Reply:
> It'll look a lot less weird if I stashed both the queue and the UrlGroup on the delegation rule.

Yes, at least do that.